### ISSUE TITLE
Transfer Learning for DagNN

### ISSUE THREAD

**Issue by ncalik** [vlfeat/autonn#27]
Hi again,
Is there any easy way for fine-tuning GoogleNet in AutoNN? 
Can I use "deepCopy" or "replace" methods? 
or should I use firstly dagnn structure to generate new nn and convert it to AutoNN?

**Comment by jotaf98** [comment]
Well the pre-trained networks are in DagNN format, so you need to convert them:

```
dag = load('imagenet-googlenet-dag.mat')
dag = dagnn.DagNN.loadobj(dag)
outputs = Layer.fromDagNN(dag)
```

This will return the outputs of the DAG as layers, which you can then pick and choose from (and move higher up in the layers to select others if you want; outputs{1}.find() usually helps).

To fine-tune you can just take the prediction layer and plug it to a new loss, or do more computations with it.

Replace is different; it's handy to do mass edits, e.g. replace all ReLUs with another activation. DeepCopy is most useful for things like siamese networks, where you want two copies of the same network to run side by side, either with shared or unshared parameters to learn.

**Comment by ncalik** [comment]
​

Your information is very useful. I will try as soon as possible
